### PR TITLE
feat(hand-tool): activate on space

### PR DESCRIPTION
**Pull request by philippfromme**
Related to camunda/camunda-modeler#321

### CODE REVIEW

File: lib/features/hand-tool/HandTool.js [hunk 1]
```diff
@@ -21,6 +26,31 @@ export default function HandTool(eventBus, canvas, dragging, toolManager) {
     }
   }, this);
 
+  eventBus.on('keyboard.keydown', HIGH_PRIORITY, function(e) {
```
Comment (barmac) [inline lib/features/hand-tool/HandTool.js]: I have two concerns:
1. Do we actually want to implement the new shortcut in the framework? Why here and not in Camunda Modeler where the issue was created?
2. Why don't we use `Keyboard#addListener` with new `type` parameter here?

Comment (philippfromme) [reply lib/features/hand-tool/HandTool.js]: 1. I discussed this with @nikku, the answer is yes, we want this to be part of the library.
2. Yeah, the code becomes a bit more verbose but makes sense.

**Comment by philippfromme** [comment]
@barmac Ready to be reviewed.

File: lib/features/hand-tool/HandTool.js [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+
 var HIGH_PRIORITY = 1500;
 var HAND_CURSOR = 'grab';
 
 
-export default function HandTool(eventBus, canvas, dragging, toolManager) {
+export default function HandTool(eventBus, canvas, dragging, keyboard, toolManager) {
   this._dragging = dragging;
 
+  var self = this;
```
Comment (barmac) [inline lib/features/hand-tool/HandTool.js]:
```suggestion
  var self = this,
      keyboard = injector.get('keyboard', false);
```

File: lib/features/hand-tool/HandTool.js [hunk 3]
```diff
@@ -1,12 +1,15 @@
 import { hasPrimaryModifier } from '../../util/Mouse';
 
+import { isKey } from '../../features/keyboard/KeyboardUtil';
+
 var HIGH_PRIORITY = 1500;
 var HAND_CURSOR = 'grab';
 
 
-export default function HandTool(eventBus, canvas, dragging, toolManager) {
+export default function HandTool(eventBus, canvas, dragging, keyboard, toolManager) {
```
[the remark this answers is not in the document]
Comment (barmac) [reply lib/features/hand-tool/HandTool.js]: Let's make `Keyboard` optional, ok?

File: lib/features/hand-tool/index.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
+import KeyboardModule from '../keyboard';
 import ToolManagerModule from '../tool-manager';
 
 import HandTool from './HandTool';
 
 export default {
   __depends__: [
+    KeyboardModule,
```
Comment (barmac) [inline lib/features/hand-tool/index.js]:
```suggestion
```

File: lib/features/hand-tool/HandTool.js [hunk 5]
```diff
@@ -21,6 +24,37 @@ export default function HandTool(eventBus, canvas, dragging, toolManager) {
     }
   }, this);
 
+  keyboard.addListener(HIGH_PRIORITY, function(e) {
```
Comment (barmac) [inline lib/features/hand-tool/HandTool.js]:
```
if (keyboard) {
...
}
```

File: lib/features/hand-tool/HandTool.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (philippfromme) [reply lib/features/hand-tool/HandTool.js]: 👍

**Comment by nikku** [comment]
:+1: :racehorse:

**Comment by nikku** [comment]
A even simpler and less verbose pattern would be:

```
keyboard && keyboard.addListennerFoo(() => {
   //  long
   // very
   // method 
   // here
});
```

**Comment by philippfromme** [comment]
@nikku Done, thx

**Comment by nikku** [comment]
Looks very great to me now. :fireworks:

**Pull request merged**